code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm

```diff
@@ -692,3 +692,33 @@
 
 	for(var/department_type in department_types)
 		create_separatist_nation(department_type, announcement = FALSE, dangerous = FALSE, message_admins = FALSE)
+
+/datum/dynamic_ruleset/roundstart/silencer
+	name = "Silencer"
+	antag_flag = ROLE_SILENCER
+	antag_datum = /datum/antagonist/silencer
+	flags = LONE_RULESET
+	minimum_required_age = 7
+	/// it's every role but mime. i'm not typing that out
```

Comment:
there should be a specific empty define like "DEFINED_PROGRAMATICALLY"

that flags to the developer this list is built at a special runtime.

Reply:
tbh I'm almost on teh fence about it and would just prefer it written out.

or add the opposite of restricted roles (allowed_roles).

Reply:
fair enough, i can implement that

Reply:
I think you should set it on `/New()`, to `typesof(/datum/job) - /datum/job/mime`

Reply:
Or whatever it is

Reply:
i did that with malf ai and it was horribly broken so idk if thatd work